lib/SILOptimizer/Mandatory/SendNonSendable.cpp

```diff
@@ -1185,106 +1187,112 @@ class BlockPartitionState {
   std::optional<PartitionOp> localInst;
 
-  static LocalConsumedReason ConsumeInst(PartitionOp localInst) {
+  static LocalTransferredReason TransferInst(PartitionOp localInst) {
     assert(localInst.getKind() == PartitionOpKind::Transfer);
-    return LocalConsumedReason(LocalConsumedReasonKind::LocalConsumeInst, localInst);
+    return LocalTransferredReason(LocalTransferredReasonKind::LocalTransferInst,
+                                  localInst);
   }
 
-  static LocalConsumedReason NonConsumeInst() {
-    return LocalConsumedReason(LocalConsumedReasonKind::LocalNonConsumeInst);
+  static LocalTransferredReason NonTransferInst() {
+    return LocalTransferredReason(
+        LocalTransferredReasonKind::LocalNonTransferInst);
   }
 
-  static LocalConsumedReason NonLocal() {
-    return LocalConsumedReason(LocalConsumedReasonKind::NonLocal);
+  static LocalTransferredReason NonLocal() {
+    return LocalTransferredReason(LocalTransferredReasonKind::NonLocal);
   }
 
   // 0-ary constructor only used in maps, where it's immediately overridden
-  LocalConsumedReason() : kind(LocalConsumedReasonKind::NonLocal) {}
+  LocalTransferredReason() : kind(LocalTransferredReasonKind::NonLocal) {}
 
 private:
-  LocalConsumedReason(LocalConsumedReasonKind kind,
-                 std::optional<PartitionOp> localInst = {})
+  LocalTransferredReason(LocalTransferredReasonKind kind,
+                         std::optional<PartitionOp> localInst = {})
       : kind(kind), localInst(localInst) {}
 };
 
 // This class captures all available information about why a value's region was
-// consumed. In particular, it contains a map `consumeOps` whose keys are
-// "distances" and whose values are Consume PartitionOps that cause the target
-// region to be consumed. Distances are (roughly) the number of times
+// transferred. In particular, it contains a map `transferOps` whose keys are
+// "distances" and whose values are Transfer PartitionOps that cause the target
+// region to be transferred. Distances are (roughly) the number of times
 // two different predecessor blocks had to have their exit partitions joined
-// together to actually cause the target region to be consumed. If a Consume
+// together to actually cause the target region to be transferred. If a Transfer
 // op only causes a target access to be invalid because of merging/joining
 // that spans many different blocks worth of control flow, it is less likely
 // to be informative, so distance is used as a heuristic to choose which
 // access sites to display in diagnostics given a racy consumption.
-class ConsumedReason {
-  std::map<unsigned, std::vector<PartitionOp>> consumeOps;
+class TransferredReason {
+  std::map<unsigned, std::vector<PartitionOp>> transferOps;
 
-  friend class ConsumeRequireAccumulator;
+  friend class TransferRequireAccumulator;
 
   bool containsOp(const PartitionOp& op) {
-    for (auto [_, vec] : consumeOps)
+    for (auto [_, vec] : transferOps)
```

Comment:
I think this and `isValid()` could be further simplified to use `llvm::any_of` and `llvm::is_contained` as a follow-up.

Reply:
@xedin TBH, I really dislike containsOp. It would be much better to just have a separate set like thing. So we can do quick PartitionOp checks. But I am going to leave it in for now since this only runs when we emit diagnostics.